pkg/cli/scan.go

```diff
@@ -606,6 +607,18 @@ func resolveInputFileFromArg(inputFilePath string) (*os.File, error) {
 	}
 }
 
+// getRepositoryURL returns the URL of the APKINDEX.tar.gz file for the given
+// repository and architecture. If the repository URL already points to an
+// APKINDEX.tar.gz file, it will be returned as-is. User input may or may not
+// have included the architecture or the APKINDEX.tar.gz suffix, so construct
+// the full URL to provide better UX.
+func getRepositoryURL(repository, arch string) string {
```

Comment:
Would it be possible to make the behavior simpler and more predictable, by requiring the caller to pass the URL to the repo and not to the index tar gz?

Reply:
Just wondering if we need to support both with and without the `.../APKINDEX.tar.gz`... It'd be simpler to say the URL has to be just to the repo, so like `https://packages.wolfi.dev/os`, instead of accepting multiple forms, unless we really need to support both?

Reply:
> if we need to support both

IIUC, `apk ls` command supports both, and some packages does provide only single architecture, thats where we may need to pass the `ARCH` in the URL.

Reply:
>some packages does provide only single architecture

If I'm following you, this problem exists with the remote scanning feature with or without this new enhancement, is that right?

Would a better solution here be to show a warning if not all architectures are found? And still error if none can be found?

I guess I'm not following how architecture availability is specific to this new flag, but maybe you can help me follow :)